src/main/kotlin/com/devfloor/untitled/article/application/ArticleService.kt

```diff
@@ -108,6 +114,14 @@ class ArticleService(
             articleHashtagRepository.deleteAllByArticle(it)
             articleFavoriteRepository.deleteAllByArticle(it)
             articleRepository.delete(it)
+            articleClickCountRepository.deleteByArticle(it)
         }
         ?: EntityNotFoundException.notExistsId(Article::class, articleId)
+
+    @Transactional
+    fun increaseHitCount(article: Article) {
+        articleClickCountRepository.findByArticleOrNull(article)
+            ?.run { ::increase }
+            ?: articleClickCountRepository.save(ArticleClickCount(article))
+    }
```

Comment:
1. 여기도 네이밍 Click으로 통일하는게 어떨까요 !?
2. 게시글 조회시에만 이 메서드가 사용 될 것 같은데, private으로 바꾸면 어떨까요!?
3. 같은 맥락으로 게시글 조회에만 사용되기 때문에 이 메소드는 항상 `showByArticleId()`에서 호출될 것 같은데, 이 때 Transactional 어노테이션 자체가 필요없을 것 같습니다.! 더 중요한건, 만약 상위 메서드인 `showByArticleId()`에 Transactional 어노테이션이 붙어있지 않다고 하더라도 이 메서드는 트랜젝셔널하게 수행되지 않아요. 왜냐하면 프록시 객체에서 내부 메서드를 호출할 때에는 해당 프록시의 메서드를 Transactioanl로 감쌌는지 알 수가 없기 때문인거로 알고 있습니다 ! 만약 이거와 관련해서 런타임에 의도한 바와 다르게 수행된다면 아마 원인을 찾기가 어려울 것 같아요!
```suggestion
    private fun increaseClickCount(article: Article) {
        articleClickCountRepository.findByArticleOrNull(article)
            ?.run { ::increase }
            ?: articleClickCountRepository.save(ArticleClickCount(article))
    }
```

Reply:
1. 통일하겠습니다 놓쳤습니다.
2. private이 좋아보입니다! 수정하겠습니다.
3. 덕분에 transactional 어노테이션과 프록시객체에 관해서 공부좀했네용! 감사합니당. 같은 클래스 내 함수 / private 함수는 transactional 이 적용되지 않네용 !